src/main/java/org/spongepowered/api/block/data/MobSpawner.java

```diff
@@ -0,0 +1,223 @@
+
+    /**
+     * Sets the number of monsters that will attempt to spawn in each batch.
+     * <p>
+     * The actual number of monsters spawned may be less than the attempted
+     * amount if the maximum number of entities allowed in the area is reached.
+     * </p>
+     * 
+     * @param count The new count
+     */
+    void setSpawnCount(short count);
+
+    /**
+     * Gets the maximum amount of entities that may be within the spawn range.
+     * This monster spawner will cease spawning new entities if this cap is
+     * reached.
+     * 
+     * @return The maximum amount of nearby entities
+     */
+    short getMaximumNearbyEntities();
+
+    /**
+     * Sets the maximum amount of entities that may be within the spawn range.
+     * This monster spawner will cease spawning new entities if this cap is
+     * reached.
+     * 
+     * @param count The new maximum amount of nearby entities
+     */
+    void setMaximumNearbyEntities(short count);
+
+    /**
+     * Gets the minimum range to the nearest player before this monster spawner
+     * will activate.
+     * 
+     * @return The required range
+     */
+    short getRequiredPlayerRange();
+
+    /**
+     * Sets the minimum range to the nearest player before this monster spawner
+     * will activate.
+     * 
+     * @param range The new required range
+     */
+    void setRequiredPlayerRange(short range);
+
+    /**
+     * Gets the range within which the monsters from each batch will be spawned.
+     * <p>
+     * The total region within which the monster may be spawned is defined by a
+     * cuboid with dimensions of {@code range*2+1 x 3 x range*2+1} centered
+     * around the monster spawner.
+     * </p>
+     * 
+     * @return The range
+     */
+    short getSpawnRange();
+
+    /**
+     * Sets the range within which the monsters from each batch will be spawned.
+     * <p>
+     * The total region within which the monster may be spawned is defined by a
+     * cuboid with dimensions of {@code range*2+1 x 3 x range*2+1} centered
+     * around the monster spawner.
+     * </p>
+     * 
+     * @param range The new range
+     */
+    void setSpawnRange(short range);
+
+    /**
+     * Sets the next entity type and properties to be spawned.
+     * 
+     * @param type The entity type
+     * @param additionalProperties Additional properties to apply to the entity
+     */
+    void setNextEntityToSpawn(EntityType type, Optional<DataContainer> additionalProperties);
+
+    /**
+     * Sets the next {@link WeightedRandomEntity} to be spawned.
+     * 
+     * @param entity The random entity entry
+     */
+    void setNextEntityToSpawn(WeightedRandomEntity entity);
+
+    /**
+     * Defines a number of {@link WeightedRandomEntity}s from which the type of
+     * each batch will be randomly selected based on the weighting value.
+     * 
+     * @param entities The possible entities
+     */
+    void setPossibleEntitiesToSpawn(WeightedRandomEntity... entities);
```

Comment:
Is there a reason why you used an var arg for the setter, but a `Collection` for the getter?

It would be nice if you add a setter using a `Collection` as well. So that you can grab the value from one spawner and set it to another spawner without converting them.